server/src/main/java/org/elasticsearch/script/field/vectors/BitMultiDenseVector.java

```diff
@@ -31,6 +31,67 @@ public void checkDimensions(int qvDims) {
         }
     }
 
+    @Override
+    public float maxSimDotProduct(float[][] query) {
+        vectorValues.reset();
+        float[] sums = new float[query.length];
+        while (vectorValues.hasNext()) {
```

Comment:
Did you implement it this way to avoid iterating over the vector values multiple times? 

However, shouldn’t we be calculating the sum of the maximums rather than the maximum of the sums? I might be misinterpreting, but this doesn’t seem to match the usual max similarity implementation.

Reply:
> Did you implement it this way to avoid iterating over the vector values multiple times?

so, we should only read the doc vector in once and compare it once against all query vectors.

> However, shouldn’t we be calculating the sum of the maximums rather than the maximum of the sums? I might be misinterpreting, but this doesn’t seem to match the usual max similarity implementation.

🤦 ugh, I did it backwards indeed. its the sum of the maximums.